build.go

```diff
@@ -158,17 +163,124 @@ func (vs *version) addFiles(partitions map[int]bool) error {
 	}
 }
 
-func (vs *version) addFile(file string, partitions map[int]bool) error {
-	if vs.stats != nil {
-		start := time.Now()
-		defer func() {
-			duration := time.Since(start)
-			tags := []string{fmt.Sprintf("sequins_db:%s", vs.db.name)}
-			vs.stats.Timing("s3.download_duration", duration, tags, 1)
-		}()
+// Download a sparkey file into a local file
+func (vs *version) sparkeyDownload(src, dst, fileType string, transform func(io.Reader) io.Reader) error {
+	success := false
+
+	disp := vs.sequins.backend.DisplayPath(vs.db.name, vs.name, src)
+	log.Printf("downloading sparkey %s file %s\n", fileType, disp)
+
+	stream, err := vs.sequins.backend.Open(vs.db.name, vs.name, src)
+	if err != nil {
+		return fmt.Errorf("opening sparkey %s file for %s: %s", fileType, disp, err)
 	}
+	defer stream.Close()
 
-	disp := vs.sequins.backend.DisplayPath(vs.db.name, vs.name, file)
+	out, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("creating sparkey %s file for %s: %s", fileType, disp, err)
+	}
+	defer func() {
+		out.Close()
+		if !success {
+			os.Remove(dst)
+		}
+	}()
+
+	var trStream io.Reader = stream
+	if transform != nil {
+		trStream = transform(trStream)
+	}
+
+	_, err = io.Copy(out, trStream)
+	if err != nil {
+		return fmt.Errorf("reading sparkey %s file %s: %s", fileType, disp, err)
+	}
+
+	success = true
+	return nil
+}
+
+// Add a Sparkey file to this version.
```

Comment:
This is great info that we should probably document outside of code (probably in the [data requirements](http://sequins.io/manual/1-2-data-requirements/) or [on disk format](http://sequins.io/manual/2-1-on-disk-format/) sections), especially since sparkey files have more strict requirements than sequence files.

Reply:
:+1: Will do.